packages/synapse-react-client/src/components/ChallengeTeamWizard/ChallengeTeamWizard.tsx

```diff
@@ -79,366 +59,368 @@ export type ChallengeTeamWizardProps = {
+      case ChallengeTeamWizardStep.JOIN_REQUEST_SENT:
+      case ChallengeTeamWizardStep.REGISTRATION_SUCCESSFUL:
+        // Cannot go back from these steps
+        return <></>
+      case ChallengeTeamWizardStep.ACCEPT_INVITATION:
+      case ChallengeTeamWizardStep.JOIN_REQUEST_FORM:
+      case ChallengeTeamWizardStep.CREATE_NEW_TEAM:
+      default:
+        // All the other steps go back to team selection
+        return (
+          <Button
+            variant={'outlined'}
+            onClick={() => {
+              setStep(ChallengeTeamWizardStep.SELECT_YOUR_CHALLENGE_TEAM)
+              setSelectedTeamId(undefined)
+            }}
+          >
+            Back
+          </Button>
         )
-        setHasSubmissionTeam(isReg)
-      }
-      setLoading(false)
-    }
-    if (userSubmissionTeamError) {
-      setErrorMessage(
-        `Error: Could not determine if you are already registered for this Challenge.`,
-      )
-      setLoading(false)
     }
-  }, [userSubmissionTeams, userSubmissionTeamError])
-
-  const { data: teamMembershipStatus, error: teamMembershipError } =
-    useGetMembershipStatus(
-      selectedTeamId ?? EMPTY_ID,
-      userProfile?.ownerId ?? EMPTY_ID,
-      {
-        enabled: !!selectedTeamId && !!userProfile,
-      },
+  }, [step])
+
+  const closeButton = useMemo(() => {
+    return (
+      <Button
+        variant={'contained'}
+        onClick={() => {
+          onClose()
+        }}
+      >
+        Close
+      </Button>
     )
+  }, [onClose])
 
-  useEffect(() => {
-    if (teamMembershipStatus) {
-      setMembershipStatus(prev => ({
-        ...prev,
-        [teamMembershipStatus.teamId]: teamMembershipStatus,
-      }))
+  // Determine modal content based on step.id
+  const { actions = <></>, content = <></> } = useMemo(() => {
+    if (!isLoggedIn) {
+      return {
+        content: (
+          <Alert severity={'error'}>
+            <SignInPrompt />
+          </Alert>
+        ),
+      }
     }
-  }, [teamMembershipStatus])
 
-  useEffect(() => {
-    if (teamMembershipError) {
-      setErrorMessage(teamMembershipError.reason)
+    if (isMemberOfSubmissionTeam) {
+      return {
+        content: (
+          <Alert severity={'error'}>
+            <Typography>
+              You are already a member of a registered submission team for this
+              Challenge.
+            </Typography>
+          </Alert>
+        ),
+      }
     }
-  }, [teamMembershipError])
-
-  /*************************
-   * React to state changes
-   ************************/
-
-  const canUserJoinTeam = () => {
-    if (selectedTeamId && selectedTeamId in membershipStatus) {
-      const {
-        canJoin,
-        membershipApprovalRequired,
-        isMember,
-        hasOpenInvitation,
-        hasOpenRequest,
-      } = membershipStatus[selectedTeamId]
 
-      if (isMember) {
-        // User cannot join this team, disable Next button
+    switch (step) {
```

Comment:
Most of the logic to drive rendering child components is now contained in this `switch`, which I think is easier to follow than a list of objects with properties that are overridden on-the-fly with effects based on local state.